src/caret_analyze/architecture/struct/message_context.py

```diff
@@ -0,0 +1,294 @@
+# Copyright 2021 Research Institute of Systems Planning, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import annotations
+
+from abc import abstractmethod
+from logging import getLogger
+from typing import Dict, Optional, Tuple
+
+from caret_analyze.value_objects.message_context import (CallbackChain, InheritUniqueStamp,
+                                                         MessageContext, MessageContextType,
+                                                         Tilde, UseLatestMessage)
+
+from .callback import CallbackStruct
+from .publisher import PublisherStruct
+from .subscription import SubscriptionStruct
+from ...exceptions import UnsupportedTypeError
+
+logger = getLogger(__name__)
+
+MessageContextType.USE_LATEST_MESSAGE = \
+        MessageContextType('use_latest_message')
+MessageContextType.INHERIT_UNIQUE_STAMP = \
+        MessageContextType('inherit_unique_stamp')
+MessageContextType.CALLBACK_CHAIN = MessageContextType('callback_chain')
+MessageContextType.TILDE = MessageContextType('tilde')
+
+
+class MessageContextStruct():
+    """Structured message context value."""
+
+    def __init__(
+        self,
+        node_name: str,
+        message_context_dict: Dict,
+        subscription: Optional[SubscriptionStruct],
+        publisher: Optional[PublisherStruct],
+        child: Optional[Tuple[CallbackStruct, ...]],
+    ) -> None:
+        # Since it is used as a value object,
+        # mutable types such as dict should not be used.
+        self._node_name = node_name
+        self.message_context_dict = message_context_dict
```

Comment:
message_context_dictはprotectedとして_を追加してください
```suggestion
        self._message_context_dict = message_context_dict
```
影響を受ける箇所も修正をお願いします。

Reply:
[7c80820](https://github.com/tier4/CARET_analyze/pull/84/commits/7c80820f9ce7401d0d41c085e43b1cae08a9d4b6)にて解決しました。